src/include/processor/operator/persistent/reader/csv/parallel_csv_reader.h

```diff
@@ -16,18 +16,23 @@ class ParallelCSVReader final : public BaseCSVReader {
 
 public:
     ParallelCSVReader(const std::string& filePath, common::CSVOption option, uint64_t numColumns,
-        main::ClientContext* context);
+        main::ClientContext* context, CSVErrorHandler* errorHandler);
 
     bool hasMoreToRead() const;
     uint64_t parseBlock(common::block_idx_t blockIdx, common::DataChunk& resultChunk) override;
     uint64_t continueBlock(common::DataChunk& resultChunk);
 
+    void reportFinishedBlock();
+
 protected:
-    void handleQuotedNewline() override;
+    bool handleQuotedNewline() override;
 
 private:
+    uint64_t getNumRowsReadInBlock() final;
     bool finishedBlock() const;
     void seekToBlockStart();
+
+    uint64_t rowsInCurrentBlock;
```

Comment:
I see a similar variable `uint64_t numRowsReadInBlock;` in serial_csv_reader.h
If they have similar usage, we should consider move it to base class.

Reply:
updated